gnomad_constraint/utils/constraint.py

```diff
@@ -195,12 +217,173 @@ def create_constraint_training_dataset(
     possible_ht = possible_ht.transmute(possible_variants=possible_ht.variant_count)
 
     # Outer join the Tables with possible variant counts and observed variant counts.
-    ht = ht.join(possible_ht, "outer")
-    # Annotate the Table with cpg site and mutation_type (one of "CpG", "non-CpG
+    ht = observed_ht.join(possible_ht, "outer")
+
+    tmp_path = new_temp_file(prefix="constraint", extension="ht")
+    ht.write(tmp_path)
+    ht = hl.read_table(tmp_path)
```

Comment:
Should I use hail's function `checkpoint` here or this is fine?

Reply:
I would use checkpoint, and then you don't need to save the name to "tmp_path" either

Reply:
i would also add a tmp dir to your `hl.init(log="/constraint_pipeline.log")` in the pipeline script like here: https://github.com/broadinstitute/gnomad_qc/blob/08974a6d0a68e3456657f0db9b6b2bf6f356de47/gnomad_qc/v4/sample_qc/assign_ancestry.py#L373

Reply:
> I would use checkpoint, and then you don't need to save the name to "tmp_path" either

Since `new_temp_file` generates different paths every time, I didn't include `overwrite` parameter. Please let me know if it's the correct way to do.

Reply:
> i would also add a tmp dir to your `hl.init(log="/constraint_pipeline.log")` in the pipeline script like here: https://github.com/broadinstitute/gnomad_qc/blob/08974a6d0a68e3456657f0db9b6b2bf6f356de47/gnomad_qc/v4/sample_qc/assign_ancestry.py#L373

Do you mean the tmp dir (`constraint_tmp_prefix`) saves the intermediate files in pipeline? I used that instead of the temp dir generated by `new_temp_file`.